src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/ExportClientGrpcResponse.cs

```diff
@@ -1,15 +1,25 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient.Grpc;
+
 namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient;
 
 internal sealed class ExportClientGrpcResponse : ExportClientResponse
 {
     public ExportClientGrpcResponse(
         bool success,
         DateTime deadlineUtc,
-        Exception? exception)
+        Exception? exception,
+        Status? status,
+        string? grpcStatusDetailsHeader)
         : base(success, deadlineUtc, exception)
     {
+        this.Status = status;
+        this.GrpcStatusDetailsHeader = grpcStatusDetailsHeader;
     }
+
+    public Status? Status { get; }
```

Comment:
Could we possibly switch this to also be non-nullable `Status`? I think it would improve the code readability to always have a status. Seems like there are only a couple spots where `null` gets passed but most paths define/pass something. We could introduce canned/static codes for the `null` ones? Not going to block for this, if you agree consider as maybe a follow-up/refactor 😄